src/Glpi/Form/ServiceCatalog/Provider/CategoryProvider.php

```diff
@@ -87,4 +87,43 @@ public function getItems(ItemRequest $item_request): array
 
         return $categories;
     }
+
+    /**
+     * @param ItemRequest $item_request
+     * @return array<array{id: int, name: string}>
+     */
+    public function getAncestors(ItemRequest $item_request): array
+    {
+        $category_id = $item_request->getCategoryID();
+        $category = Category::getById($category_id);
+        if (!$category) {
+            return [];
+        }
+
+        $categories = [];
+        $current_category = [
+            'id' => $category->getID(),
+            'name' => $category->fields['name'],
+        ];
+
+        $ancestors_iterator = $category->getAncestors();
+        if (!$ancestors_iterator) {
+            return [$current_category];
+        }
+
+        /** @var Category[] $ancestors */
+        $ancestors = iterator_to_array($category->getAncestors());
```

Comment:
It will prevent to request them twice.

```suggestion
        $ancestors = iterator_to_array($ancestors_iterator);
```